FinSearchUnified/FinSearchUnified.php

```diff
@@ -19,4 +22,45 @@ public function build(ContainerBuilder $container)
         parent::build($container);
         $container->setParameter('fin_search_unified.plugin_dir', $this->getPath());
     }
+
+    public function deactivate(DeactivateContext $context)
+    {
+        $this->deactivateCustomizedPlugin();
+        parent::deactivate($context);
+    }
+
+    public function uninstall(UninstallContext $context)
+    {
+        $this->deactivateCustomizedPlugin();
+        parent::uninstall($context);
+    }
+
+    /**
+     * Try to deactivate any customization plugins of FINDOLOGIC
+     */
+    private function deactivateCustomizedPlugin()
+    {
+        /** @var InstallerService $pluginManager */
+        $pluginManager = $this->container->get('shopware_plugininstaller.plugin_manager');
+
+        try {
+            /** @var \Shopware\Models\Plugin\Plugin $plugin */
+            $plugin = $pluginManager->getPluginByName('ExtendFinSearchUnified');
+            if ($plugin->getActive()) {
+                $pluginManager->deactivatePlugin($plugin);
+            }
+        } catch (\Exception $exception) {
+            Shopware()->PluginLogger()->info('ExtendFinSearchUnified plugin doesn\'t exist!');
+        }
+
+        try {
+            /** @var \Shopware\Models\Plugin\Plugin $plugin */
+            $plugin = $pluginManager->getPluginByName('FinSearchCustomized');
```

Comment:
This whole block may be removed since we agreed on leaving the name as is to avoid legacy plugins

Reply:
Done!